torch/csrc/jit/passes/onnx/prepare_inplace_ops_for_onnx.cpp

```diff
@@ -0,0 +1,60 @@
+#include <torch/csrc/jit/passes/onnx/prepare_inplace_ops_for_onnx.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+// aten::pop is inplace. The tensor list input is updated.
+// This pass creates an aten::__getitem__ op to return the original output from aten::pop.
+// Then it makes the original aten::pop operator return the updated tensor list,
+// and replaces all later uses of that tensor list with this new output.
+static void PrepareListPopForONNX(Block* b) {
+  for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end; ++it) {
+    for (auto* child_block : it->blocks()) {
+      PrepareListPopForONNX(child_block);
+    }
+
+    if (it->kind() == aten::pop) {
+      //   %ten : Tensor = aten::pop(%seq, %pos)
+      // Convert to
+      //   %ten : Tensor = aten::__getitem__(%seq, %pos)
+      //   %new_seq : Tensor[] = aten::pop(%seq, %pos)
+      // And replace all uses of %seq afterwards with %new_seq
+      Node* getitem_node =
+          b->owningGraph()->create(aten::__getitem__, {it->inputs()});
+      getitem_node->output()->copyMetadata(it->output());
+      getitem_node->insertBefore(*it);
+      it->output()->replaceAllUsesWith(getitem_node->output());
+
+      it->output()->copyMetadata(it->inputs()[0]);
+      it->inputs()[0]->replaceAllUsesAfterNodeWith(*it, it->output());
+    }
+  }
+}
+
+static void PrepareListAppendInsertForONNX(Block *b) {
+  for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end; ++it) {
+    for (auto* child_block : it->blocks()) {
+      PrepareListPopForONNX(child_block);
```

Comment:
Do you mean PrepareListAppendForONNX here?

Reply:
Changing to `PrepareListAppendAndInsertForONNX` ...